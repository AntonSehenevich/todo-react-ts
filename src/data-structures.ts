export enum TodoFilter {
  All = 1,
  Completed,
  Uncompleted
}

export type Todo = {
  id: string
  text: string
  completed: boolean
}

export type TodoList = Todo[]

export type EditTodoParams = Pick<Todo, 'id' | 'text'>

export type ReorderTodosParams = {
  source: string
  destination: string
}
