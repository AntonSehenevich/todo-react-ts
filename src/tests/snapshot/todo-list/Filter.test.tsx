import renderer from 'react-test-renderer'
import Filter from '../../../components/todos/filter'
import { TodoFilter } from '../../../data-structures'

test('renders correctly', () => {
  const tree = renderer
    .create(<Filter filter={TodoFilter.All} setFilter={jest.fn()} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
