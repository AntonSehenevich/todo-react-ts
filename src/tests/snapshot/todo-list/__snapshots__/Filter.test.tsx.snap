// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="list"
  data-testid="todos-filter"
>
  <li>
    <button
      className="button active"
      onClick={[Function]}
      type="button"
    >
      All
    </button>
  </li>
  <li>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Active
    </button>
  </li>
  <li>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Completed
    </button>
  </li>
</ul>
`;
